### ISSUE TITLE
Feature Request: find all Storyboard UI elements for which an AccessibilityID has not been set

### ISSUE THREAD

**Issue by seclorum** [fastlane/fastlane#11878]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines]
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

I would love to have a feature in Fastlane which parses the Storyboards for a project and produces a report of which UI elements do not have AccessibilityID set .. this would make it easier to send to the Testing Engineer, who can easily see what elements are not yet named (and which are therefore not accessible to a UITest) ...

**Comment by janpio** [comment]
Interesting idea. Do you have an idea which tool to place this in? Where does it belong to best?

**Comment by minuscorp** [comment]
I feel that this might be more on the side of a `Build Phase` script through a library that checks the storyboards of a project and generate warnings than on the _fastlane_ suite.

**Comment by janpio** [comment]
Don't you think a "good code" or "accesibility check" would be a good addition to fastlane? I do, I just can't think of a good place to put it right now.

**Comment by seclorum** [comment]
Wouldn't it be a sub-function of snapshot?  Seems to me that automatic screenshot generation, as well as automated reporting of Accessibility, are kind of along similar lines?  I haven't grok'ed the fastlane snapshot code well enough yet, but I would imagine it'd be aligned with the existing automated screenshot methods currently used?  i.e. parsing Storyboards.

**Comment by janpio** [comment]
Snapshot runs tests and takes screenshots - pretty different task I think.

**Comment by seclorum** [comment]
Fair enough.  I guess I'll have to spend a few days getting up to speed with the fastlane code base to come up with a recommendation on where to put such a feature .. I guess I could be referred to as an "Accessibility Lint" style function ..

**Comment by seclorum** [comment]
Any chance I could get a statement from you guys on if you think this feature will end up on the workbench in the near future?  If its not something of interest, I'll probably try to hack something together myself else wise, but if you think it is worth it - but don't have the time - then I might invest into grok'ing the code in fast lane well enough to send a PR ..

**Comment by janpio** [comment]
Maybe as a fastlane plugin? That would totally make sense.

That way the actual logic could be implemented and tested without any commitment needed from the core contributors, and if it turns out to be super popular and used very often it can maybe still be integrated to the core action further down the line.

You can read more about creating a plugin here: https://docs.fastlane.tools/plugins/create-plugin/
